src/huggingface_hub/hf_file_system.py

```diff
@@ -649,6 +654,58 @@ def _upload_chunk(self, final: bool = False) -> None:
             )
 
 
+class HfFileSystemStreamFile(fsspec.spec.AbstractBufferedFile):
+    def __init__(
+        self, fs: HfFileSystem, path: str, revision: Optional[str] = None, block_size=0, cache_type="none", **kwargs
+    ):
+        if block_size != 0:
+            raise ValueError(f"HfFileSystemStreamFile only supports block_size=0 but got {block_size}")
+        if cache_type != "none":
+            raise ValueError(f"HfFileSystemStreamFile only supports cache_type='none' but got {cache_type}")
+        try:
+            self.resolved_path = fs.resolve_path(path, revision=revision)
+        except FileNotFoundError as e:
+            if "w" in kwargs.get("mode", ""):
+                raise FileNotFoundError(
+                    f"{e}.\nMake sure the repository and revision exist before writing data."
+                ) from e
+        self.details = {"name": self.resolved_path.unresolve(), "size": None}
```

Comment:
out of curiosity, what is it for?

Reply:
It's used to avoid an unecessary `.info()` call to instantiate `.details` - I'll add a comment